src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -468,19 +513,23 @@ FILE* Compiler::fgOpenFlowGraphFile(bool* wbDontClose, Phases phase, PhasePositi
 
 #ifdef DEBUG
     dumpFunction = JitConfig.JitDumpFg().contains(info.compMethodHnd, info.compClassHnd, &info.compMethodInfo->args);
-    filename     = JitConfig.JitDumpFgFile();
-    pathname     = JitConfig.JitDumpFgDir();
-
-    prePhasePattern  = JitConfig.JitDumpFgPrePhase();
-    postPhasePattern = JitConfig.JitDumpFgPhase();
+    CharRelease filenameMem;
+    CharRelease pathnameMem;
+    filename     = NarrowString(JitConfig.JitDumpFgFile(), filenameMem);
```

Comment:
A nit - maybe we could use `alloca` and implement the `NarrowString` as a macro to get rid of the need of the `CharRelease` stuff?

Reply:
Using `alloca` is a good idea. Not sure on the macro approach, does the JIT team have any concerns with macros?

Reply:
Another option (which would be more in line with how the rest of similar Jit code operates) would be to allocate the strings using the debug allocator `getAllocatorDebugOnly().allocate<char>(count)` and not worry about `free`ing them.

Reply:
Oh nice! I didn't realize that exists. Done.